web-ng/src/app/shared/models/form/field.model.ts

```diff
@@ -39,6 +39,8 @@ export enum FieldType {
   TEXT = 1,
   MULTIPLE_CHOICE = 2,
   PHOTO = 3,
+  DATE = 4,
```

Comment:
Please also ensure that both of these datatypes contain timezone info OR the underlying transport format is epoch time.

Reply:
for this need to make sure that the input coming from android form should be in epoch format not in string format will look into it.

Reply:
> for this need to make sure that the input coming from android form should be in epoch format not in string format will look into it.

if you use epoch format - you need to validate it is an uint64 - and the enum should have units specified (seconds, milliseconds etc as _S, _MS, etc)

Reply:
would be storing date and time in epoch format. for example (1617972920290)  https://developer.android.com/reference/java/lang/System#currentTimeMillis()

Reply:
will need to make some changes in web as well so that when the user views it is should converted to human readable format and when import csv is called it should be in epoch format.

Reply:
epoch is usually used for date + time in ms. In the case of `DATE`, we're only storing the date, not the time. 

Perhaps ISO date format string is more appropriate (`YYYY-MM-DD`)?  We don't need to bind our data model on how things are persisted in local/remote db, but worth considering that for convenience our data model is influenced by Firestore, which stores data in a quasi-JSON-like structure. In these structures, ISO 8601 appears to be the de facto standard:
https://docs.jsonata.org/date-time#json-and-iso-8601

In those case `YYYY-MM-DD` is acceptable to date-only fields.

Reply:
hmm, some comments, 
a)  downstream it is VERY important to have timezone info. Manually mucking with timezones and trying to match satellite passover times with sensor readings and ground observations etc is too complicated to deal with if done in a non-standard way. 

b) `DATE` I am ok being `YYYY-MM-DD Z`  but that doesn't conform with ISO 8601 format.  https://stackoverflow.com/questions/52215961/how-can-i-write-an-iso-8601-date-with-a-timezone-but-no-time-component#:~:text=Officially%2C%20the%20ISO%208601%20spec,intervals%20%2D%20mentioned%20further%20below).

c) `TIME` that we need for our project isn't time of day - but rather `DATETIME` (with timezone). So again having epoch time here makes life easy. (our usecase is time of irrigation start and stop - sometime it could start at 11 pm to 1 am - not normal, but somebody could it) 

My recommendation is - The UI representation should be decided on the Android side / web frontend side, but storage should be one of epoch time or timezone aware datetime. Keeping both `DATE` and `TIME` as one of those 2 conveys information to UI how to interpret it for UI but it stores using the same format for both. Keeping things as epoch time - means it is not human readable in storage, but we significantly reduce opportunities of making mistakes by potentially not assigning timestamp etc and doing some computation with it. (also storage space is reduce - n character string vs uint64)